vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantToVariantContextConverter.java

```diff
@@ -131,16 +131,14 @@ private String transformConsequenceTypeToCsqTag(String allele, ConsequenceType c
         Integer cDnaPosition = consequenceType.getcDnaPosition();
         Integer cdsPosition = consequenceType.getCdsPosition();
 
-        StringBuilder csqSb = new StringBuilder();
-        csqSb.append(allele != null ? allele : "").append("|")
-                .append(soNames != null ? soNames : "").append("|")
-                .append(symbol != null ? symbol : "").append("|")
-                .append(gene != null ? gene : "").append("|")
-                .append(feature != null ? feature : "").append("|")
-                .append(bioType != null ? bioType : "").append("|")
-                .append(cDnaPosition != null ? cDnaPosition : "").append("|")
-                .append(cdsPosition != null ? cdsPosition : "");
-        return csqSb.toString();
+        return (allele != null ? allele : "") + "|" +
+                (soNames != null ? soNames : "") + "|" +
+                (symbol != null ? symbol : "") + "|" +
+                (gene != null ? gene : "") + "|" +
+                (feature != null ? feature : "") + "|" +
+                (bioType != null ? bioType : "") + "|" +
+                (cDnaPosition != null ? cDnaPosition : "") + "|" +
+                (cdsPosition != null ? cdsPosition : "");
```

Comment:
This is not what I meant with my last comment. Having so many strings concatenated with the + operator will force the JVM to create **a lot** of them and will saturate the memory.

What I meant was something like:

```java
String symbol = consequenceType.getGeneName() != null ? consequenceType.getGeneName() : "";
String gene = consequenceType.getEnsemblGeneId() != null ? consequenceType.getEnsemblGeneId() : "";

csqSb.append(symbol).append("|").append(gene).append("|");
```

It may slow down the execution a bit though, because the getters will be called twice. What do others think? Shall we revert the code to what it was or use this suggestion?

Reply:
I prefer the previous code, but I'm fine with `String symbol = consequenceType.getGeneName() != null ? consequenceType.getGeneName() : ""; ... csqSb.append(symbol).append("|")`.

Agree that plain concatenation is not the best idea